crates/driver/src/domain/competition/quote.rs

```diff
@@ -0,0 +1,182 @@
+use {
+    crate::{
+        domain::{competition, eth},
+        infra::solver::{self, Solver},
+        util,
+    },
+    itertools::Itertools,
+};
+
+#[derive(Debug)]
+pub struct Config {
+    pub optimal_timeout: std::time::Duration,
+    pub fast_timeout: std::time::Duration,
+}
+
+/// A quote describing the expected outcome of an order.
+#[derive(Debug)]
+pub struct Quote {
+    pub sell: eth::Asset,
+    pub buy: eth::Asset,
+    pub interactions: Vec<competition::solution::Interaction>,
+}
+
+impl Quote {
+    fn new(
+        fulfillment: competition::solution::trade::Fulfillment,
+        interactions: Vec<competition::solution::Interaction>,
+    ) -> Self {
+        match fulfillment.order.side {
+            competition::order::Side::Buy => Self {
+                sell: fulfillment.order.sell,
+                buy: fulfillment.executed.to_asset(&fulfillment.order),
+                interactions,
+            },
+            competition::order::Side::Sell => Self {
+                sell: fulfillment.executed.to_asset(&fulfillment.order),
+                buy: fulfillment.order.buy,
+                interactions,
+            },
+        }
+    }
+}
+
+/// An order which needs to be quoted.
+#[derive(Debug)]
+pub struct Order {
```

Comment:
Not a fan of the name `Order` as it feels like its not a related concept to an order. This is conceptually more related to [price quoting](https://www.investopedia.com/terms/q/quoted-price.asp), but I can't find a good name for the "quote request" anywhere.

Reply:
No strong feelings, so if you think this is the better name, then we can roll with it.